### PR TITLE
simplify background tasks with tokio

### PR DESCRIPTION
**Pull request by kvc0**
Previously goodmetrics_rs had been tokio-agnostic, but it's cleaner, easier, faster, and more obvious how to use it this way.

### CODE REVIEW

File: lib/benches/benchmarks/aggregation.rs [hunk 1]
```diff
@@ -14,6 +13,7 @@ use goodmetrics::{
         stream_sink::StreamSink,
     },
 };
+use tokio::sync::mpsc;
```
Comment (tylerburdsall) [inline lib/benches/benchmarks/aggregation.rs:16]: If the std library is being thrown out, would you consider using flume? Flume tends to yield even better performance than tokio

File: lib/src/downstream/goodmetrics_downstream.rs [hunk 2]
```diff
@@ -20,37 +21,43 @@ use crate::{
...
 
-    pub async fn send_batches_forever(&mut self, receiver: Receiver<Vec<Datum>>) {
+    pub async fn send_batches_forever(mut self, mut receiver: mpsc::Receiver<Vec<Datum>>) {
         let mut interval = tokio::time::interval(Duration::from_millis(500));
         interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
         loop {
             interval.tick().await;
             // Send as quickly as possible while there are more batches
-            while let Ok(batch) = receiver.try_recv() {
+            while let Some(batch) = receiver.recv().await {
```
Comment (tylerburdsall) [inline lib/src/downstream/goodmetrics_downstream.rs:60]: Oh yeah, this is definitely going to be better.

**Review by tylerburdsall — COMMENTED** [review]
Left only one question, I would expect criterion would ultimately answer it but it's not something I had the ability to test late at night

File: lib/benches/benchmarks/aggregation.rs (hunk 1 above)
Comment (kvc0) [reply lib/benches/benchmarks/aggregation.rs:16]: That would need a compelling benchmark - it's another dependency, and this library is probably already too wide with Tokio.

**Pull request merged**